Sources/Moya/TargetType.swift

```diff
@@ -13,11 +13,8 @@ public protocol TargetType {
     /// The HTTP method used in the request.
     var method: Moya.Method { get }
 
-    /// The parameters to be encoded in the request.
-    var parameters: [String: Any]? { get }
-
-    /// The method used for parameter encoding.
-    var parameterEncoding: ParameterEncoding { get }
+    /// The default parameterEncoding for the `.encoded` RequestDataType case.
```

Comment:
Did as suggested. Two variations though. One: I used `ParameterEncoding` as the return type isntead of `JSONEncoding`. Two: I had to backtick `default` as I saw an error (in Xcode 9 beta 3).